src/components/dropdown-item/daikin-dropdown-item.ts

```diff
@@ -89,38 +104,94 @@ export class DaikinDropdownItem extends LitElement {
   @property({ type: Boolean, reflect: true })
   selected = false;
 
-  @query("button")
-  private _button!: HTMLButtonElement | null;
+  @property({ type: Boolean, reflect: true })
+  selectable = false;
+
+  @query("button,span[role=option]")
+  private _focusableElement!: HTMLElement | null;
+
+  private _handleClick(event: PointerEvent): void {
+    if (
+      this.disabled ||
+      (event.target as HTMLElement).tagName === "DAIKIN-CHECKBOX"
+    ) {
+      return;
+    }
+
+    this._emitSelect();
+    event.preventDefault();
+  }
+
+  private _handleChange(event: Event): void {
+    event.preventDefault();
+    this._emitSelect();
+  }
 
-  private _handleClick(): void {
-    if (this.disabled) {
+  private _handleKeydown(event: KeyboardEvent) {
+    if (!isSimilarToClick(event.key) || this.disabled) {
       return;
     }
 
-    this.dispatchEvent(
-      new Event("select", {
-        bubbles: true,
-      })
-    );
+    this._emitSelect();
+  }
+
+  private _handleMousedown(event: MouseEvent) {
+    if (event.detail === 2) {
+      // Prevent text selection on double click.
+      event.preventDefault();
+    }
+  }
+
+  private _emitSelect(): void {
+    this.dispatchEvent(new Event("select", { bubbles: true }));
   }
 
   override render() {
-    return html`<button
-      type="button"
-      class=${cvaOption({ selected: this.selected })}
-      data-value=${this.value}
-      role="option"
-      aria-selected=${this.selected}
-      tabindex="-1"
-      ?disabled=${this.disabled}
-      @click=${this._handleClick}
-    >
-      <slot></slot>
-    </button>`;
+    const optionClassName = cvaOption({
+      selected: this.selected,
+      disabled: this.disabled,
+      multiple: this.selectable,
+    });
+
+    return this.selectable
+      ? html`<span
+          class=${optionClassName}
+          data-value=${this.value}
+          role="option"
+          aria-selected=${this.selected}
+          tabindex="-1"
+          aria-disabled=${this.disabled}
+          @click=${this._handleClick}
+          @keydown=${this._handleKeydown}
+          @mousedown=${this._handleMousedown}
+        >
+          <daikin-checkbox
+            check-state=${this.selected ? "checked" : "unchecked"}
+            label=${this.textContent ?? ""}
```

Comment:
Actually this is not correct. We need to track changes to the contents of the slot, but we can't do that technically.
We may need to change the design of the checkbox so that it can receive a slot.

It looks like we'll need to leave this as an issue for now.

Reply:
I think we should use `slot.textContent` instead of `this.textContent`.

```suggestion
            label=${
            /* FIXME: Stop using `textContent` to follow the updates to the text. Rather, we should change checkbox to accept a slot. */
            slot.textContent ?? ""}
```